app/javascript/packs/restaurantsViewScreen/reastaurantsScreen.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Restaurants from './restaurants';
+import { restaurantsProvider } from '../../services/DataProvider';
+import SelectionFilters from './selectionFilters';
+import RestaurantsGoogleMap from './restaurantsMap';
+
+export default class RestaurantsScreen extends React.Component {
+  state = {
+    restaurants: [],
+    filtered: [],
+    rating: 0,
+    cuisine: 'all',
+    maximumDeliveryTime: 'all',
+  };
+
+  setRestaurants = (restaurants) => {
+    this.setState({ restaurants }, this.filter);
+  };
+
+  componentWillMount () {
+    restaurantsProvider.get('/restaurants')
+      .then(this.setRestaurants)
+      .catch(console.error);
+  }
+
+  filter = () => {
```

Comment:
you're using a lot of properties from `state`, you may as well destructure the ones you need at the beginging:

``` 
const { resturants, rating, cuisine, .. } = this.state 
...
const isCuisineEligible = cuisine === 'all' || cuisine === restaurant.cuisine.name;
 

```